### PR TITLE
[docs] Improve doc readability by changing the theme and style

### PR DESCRIPTION
**Pull request by mikesnoeren**
## The Problem/Issue/Bug:
The readability within the new docs should be improved;
1. The light-blue theme has a very low contrast ratio which results in poor accessibility
2. The line length within the docs is too long, which makes text harder to read and understand.
3. Heading margin feels off, the content related to the heading is too far away.
4. Heading styling should stand out more, this in terms helps with navigation throughout the page.
5. Blockquotes should stand out more, they usually contain important information the user has to read. Currently it can easily be overlooked,
 
## How this PR Solves The Problem:
1. The light-blue theme has been replaced with indigo, as the contrast between backgrounds and text is better.
2. The max width for `div.md-content` has been reduced by almost 300px.
3. All heading margins have been reduced and are now being set relative to the headings font-size
4. Headings are now bold and their color is set to the primary theme color so you distinguish between content and headings better.
5. Blockquotes styling has changed

## Manual Testing Instructions:
You can view this PR's results by using [this link](https://ddev--4009.org.readthedocs.build/en/4009/).

<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/4009"></a>

### CODE REVIEW

**Comment by rfay** [comment]
I notice the increased size of the tabs makes lots of them disappear:
DDEV_Installation_-_DDEV_Docs

**Comment by mikesnoeren** [comment]
There are a couple things we could do to fix the disappearing tabs which I will list down below, but keep in mind that scrollable tabs is the expected behaviour when they don't fit next to each other, I therefore would opt for a content solution in stead of an styling solution.

- **Make the tab buttons smaller again**
I do not recommend that as the text within the buttons was (in my opinion) too small and didn't fit with the other UI elements.
- **Tabs wrap to the next line**
Not my favourite solution, but with a few style changes it's certainly possible. The result could look somewhat like the images below:
  <details> 
    <summary>Flex wrap tabs example images</summary>
      Screenshot 2022-07-17 at 13 14 34
      Screenshot 2022-07-17 at 13 14 43
  </details>

- **Rethink the content structure**
Whilst the use of tabs is a massive improvement over just putting everything on a page, we could/should look into changing the content structure a bit;
  - **Combine some tabs**
  Example: the tabs "Windows WSL2" and "Traditional Windows" could live under a tab named "Windows". Within this tab content we can use headers to split them back into two seperate sections.
  - **Make a subpage for each OS** 
  If we split the content in pages (example: Windows, Mac, Linux and perhaps 'Alternative options'), we would have less tabs per page which would make them fit more easily. This is a very popular approach, a lot of big docs do it that way. (example: [Docker Desktop docs](https://docs.docker.com/desktop/install/mac-install/))

**Comment by rfay** [comment]
My eyes are bad and I wasn't fussing about the tabs the way they were :) 

The problem with making it so only a few tabs show on the page is that it makes the overall options less likely to be discovered, but yes, the horizontal scrolling is anticipated by everybody I think. 

Content structure change with subpage for each OS might work.

**Comment by shaal** [comment]
I'm not sure if mkdocs supports this but could be a nice interface to have a select list "choose your OS", clicking on it and you serl all the options. Whatever you select is saved as a cookie, and every time that interface is used, it's using your last selection as default.

**Comment by mikesnoeren** [comment]
> I'm not sure if mkdocs supports this but could be a nice interface to have a select list "choose your OS", clicking on it and you serl all the options. Whatever you select is saved as a cookie, and every time that interface is used, it's using your last selection as default.

Was thinking about something similar but with the initial select being automatic as we can use JS to check the users OS. 
After some thinking I decided not to add this to my last comment as it potentially adds extra steps for the user to get what they want, and I figure a lot of people using DDEV clear their browser cache regularly anyways.

**Comment by rfay** [comment]
> Was thinking about something similar but with the initial select being automatic as we can use JS to check the users OS.
After some thinking I decided not to add this to my last comment as it potentially adds extra steps for the user to get what they want, and I figure a lot of people using DDEV clear their browser cache/cookies regularly anyways.

It might be possible to set defaults using js to choose a default tab, but of course, not to hide all the other options. But remember that many of the tab options are not just about OSs. And there's not just one way to install things on a particular OS.

**Comment by shaal** [comment]
Sorry, I meant all options, not just OS.

**Comment by rfay** [comment]
In https://github.com/drud/ddev/pull/4015 I consolidated the docker and ddev installations by OS. See https://ddev--4015.org.readthedocs.build/en/4015/

**Comment by mikesnoeren** [comment]
> I hope you've been following
> 
> * [Links to quickstart guides do not work as expected #4019](https://github.com/drud/ddev/issues/4019)
>   It looks like tabs may have to come back out :(

I hope a fix for this will be implemented soon, I do however see two possible outcomes for the tab linking issue:
1. We could make a temporarily fix that uses JS to open the right tabs, until this issue gets externally fixed.
2. We split our content over seperate pages. I think this is my preferred option as less text present on a page usually translates to better readability and understandability..

That being said: I don't think this directly interferes with this PR?

**Comment by rfay** [comment]
The lead two items of your PR were about tabs, so it's very important, since we might have to rip tabs out.

If you can find a way to use js and solve the general problems we have here... it would be awesome!

**Comment by mikesnoeren** [comment]
With the tabs 'fixed' for now (I will still continue improving them), what's the status of this PR? Is there still testing required?

**Comment by rfay** [comment]
Please rebase it so it reflects the current setup, then we need to get some eyeballs on it. I don't think I'll be a fan of the bigger tabs, but OTOH I did a lot of work to reduce the number of tabs when they were too many.

**Comment by rfay** [comment]
When you rebase we can all take a look, thanks! If you need help with that, happy to help. I don't trust Github's UI, but if you want to try it you can.

What I do:

```
git fetch upstream
git rebase upstream/master
```

If merge conflicts happen you have to use a `git mergetool` or any other tool include poking into the textual merge conflicts.

Here's github's UI, you'd want to "update with rebase" if possible.

_docs__Improve_doc_readability_by_changing_the_theme_and_style_by_mikesnoeren_·_Pull_Request__4009_·_drud_ddev

**Comment by mikesnoeren** [comment]
> When you rebase we can all take a look, thanks! If you need help with that, happy to help. I don't trust Github's UI, but if you want to try it you can.
> 
> What I do:
> 
> ```
> git fetch upstream
> git rebase upstream/master
> ```
> 
> If merge conflicts happen you have to use a `git mergetool` or any other tool include poking into the textual merge conflicts.
> 
> Here's github's UI, you'd want to "update with rebase" if possible.
> 
> _docs__Improve_doc_readability_by_changing_the_theme_and_style_by_mikesnoeren_·_Pull_Request__4009_·_drud_ddev

Thanks for that, I just rebased.

**Comment by rfay** [comment]
I like it. It's also better now that the too-many-tab ones have less tabs. Asked for comment in https://discord.com/channels/664580571770388500/1006667454019276812

File: docs/content/assets/extra.css [hunk 1]
```diff
@@ -1,23 +1,70 @@
...
+}
+
+[dir=ltr] .md-typeset blockquote p:last-child {
+    margin-bottom: 0;
+}
+
+/* tabs */
+.md-typeset .tabbed-labels label {
+    font-size: 16px;
+}
```
Comment (gilbertsoft) [inline docs/content/assets/extra.css:70]: Please don't use pixels here but rem instead.

```suggestion
    font-size: 0.75rem;
}

```

**Review by gilbertsoft — CHANGES_REQUESTED** [review]
I'd also suggest to reduce the font size in general a bit from 0.8 rem to 0.75.

The constrast and readability improvement is great, thanks!

**Comment by rpkoller** [comment]
Thank you for working on this! I've briefly tested the [link](https://ddev--4009.org.readthedocs.build/en/4009/) @rfay posted over on the ddev discord yesterday. It is already a huge improvement over the previous version. A few details i've noticed: 

- the none active menu items have a color contrast of `4.27:1` but the goal should be at least `4.5:1` ( WCAG SC 1.4.3)
- the permalink icons (not on mouse over) have a color contrast of `2.23:1`
- the indigo color of the active tab of the system requirements section has a color contrast of `4.26:1`
- the line length was reduced from ~120-125 to ~95-100 characters. i would perhaps even reduce it a bit further. even though it is a AAA success criterion but 1.4.8 has a limit of 80 characters (https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-visual-presentation.html). i have personally issues with too long line lengths cognitively. the old version was definitely too long and challenging while the new version is an improvement but still not easy keeping the line and jumping from one to another. 
- and would it make sense to underline links? that way it would be more easy to distinguish them from regular text (that way the links wouldn't solely be labeled/identified by color)

**Comment by rfay** [comment]
This isn't necessarily a free-for-all about what everybody would do to the docs. Would it make sense to get this in and then do an accessibility PR @rpkoller ? So glad you know about such things!

**Comment by shaal** [comment]
Main menu items do not have enough color contrast.
image

We should update them to be same color, and mark the active one with underline or something like that

**Comment by ouw-jvt** [comment]
My initial response is extremely positive. The reduced line length really helps! The improved contrast of the blue, along with the type treatment on headings, really nice as well.
---

Maybe this will help on the tabs - here's an adjustment I mocked up in dev tools.

image

- More accurately vertical centering on the arrows
- Larger tap area for mobile
- Better contrast against the text behind it

gif comparing the two

https://share.cleanshot.com/DP54pe


I'd be happy to submit a quick PR on that if you'd like

**Comment by mikesnoeren** [comment]
Thank you for all the feedback! I think the given suggestions here are great and I will implement them as soon as I have time to do so..

**Comment by rfay** [comment]
When you rebase this the windows-nfs failure will go away. Finally fixed that.

**Comment by mikesnoeren** [comment]
I just made the following changes;
- Adjusted colors used in the main menu, it is now clear on what page you are.
- Increase the tab button sizes for smaller screens as they were barely clickable.

**Comment by rfay** [comment]
I spent some time with this and it looks great to me. Let me know when you're ready to have it merged.

**Review by rfay — APPROVED** [review]

**Comment by mikesnoeren** [comment]
I am ready to have it merged, if any problems arrise issues can be made and I'll fix those later.

**Comment by rfay** [comment]
Awesome, thanks so much! You've made huge contributions here and elsewhere. The navigation is working great for me.

**Pull request merged**